### PR TITLE
Explore places filter - Second merge

### PR DESCRIPTION
**Pull request by asogwachiuba**
Updated explore places filter logic

### CODE REVIEW

File: lib/features/places/views/explore_places/explore_places_viewmodel.dart [hunk 1]
```diff
@@ -131,20 +131,27 @@ class ExplorePlacesViewModel extends IbloovViewModel {
...
+              return PlacesCategory.fromJson(category).id ==
+                  selectedCategory.id;
             }
           },
-          orElse: () => EventCategory(),
+          orElse: () => null,
         );
         // If there is a similar category, then it mean's the Place model has the selected category
-        if (category != null) {
+        if (category is String && category.isNotEmpty) {
```
Comment (divincoder) [inline lib/features/places/views/explore_places/explore_places_viewmodel.dart:150]: category can be null to remove possibilities of null pointer incase null is not a type in Dart. Add a null check here

**Comment by divincoder** [comment]
Attend to comment and resolve conflicts

File: lib/features/places/views/explore_places/explore_places_viewmodel.dart (hunk 1 above)
Comment (asogwachiuba) [reply lib/features/places/views/explore_places/explore_places_viewmodel.dart:150]: Line 147 null checks category and then other checks are carried out if it is not null. Hope this satisfies the null check required.

**Comment by asogwachiuba** [comment]
> Attend to comment and resolve conflicts
Okay on it

**Review by divincoder — APPROVED** [review]

**Pull request merged**